src/alert/alert.spec.ts

```diff
@@ -99,6 +99,29 @@ describe('NgbDismissibleAlert', () => {
            });
      })));
 
+  it('should auto close on timeout specified', fakeAsync(inject([TestComponentBuilder], (tcb) => {
+       const html = '<template ngbAlert [dismissOnTimeout]="1000">Hello, {{name}}!</template>';
+       tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+         fixture.detectChanges();
+         const alertEl = getAlertElement(fixture.nativeElement);
+         fixture.detectChanges();
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(1000);
+         fixture.detectChanges();
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(1200);
+         fixture.detectChanges();
+         expect(getAlertElement(fixture.nativeElement)).toBeNull();
+
+         discardPeriodicTasks();
```

Comment:
`discardPeriodicTasks` shouldn't be needed here as at this point all the periodic tasks should be gone.